app/src/components/market/common/upgrade_proxy/index.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { DOMAttributes, HTMLAttributes } from 'react'
+import styled from 'styled-components'
+
+import { ConnectedWeb3Context } from '../../../../hooks'
+import { getNativeAsset } from '../../../../util/networks'
+import { ButtonStateful, ButtonStates } from '../../../button/button_stateful'
+
+const Wrapper = styled.div`
+  border-radius: 4px;
+  border: ${({ theme }) => theme.borders.borderLineDisabled};
+  padding: 21px 25px;
+`
+
+const Title = styled.h2`
+  color: ${props => props.theme.colors.textColorDark};
+  font-size: 16px;
+  letter-spacing: 0.4px;
+  line-height: 1.2;
+  margin: 0 0 20px;
+  font-weight: 400;
+`
+
+const DescriptionWrapper = styled.div`
+  align-items: center;
+  display: flex;
+`
+
+const Description = styled.p`
+  color: ${props => props.theme.colors.textColorLightish};
+  font-size: 14px;
+  letter-spacing: 0.2px;
+  line-height: 1.4;
+  margin: 0 32px 0 0;
+`
+
+export type UpgradeProxyProps = DOMAttributes<HTMLDivElement> &
+  HTMLAttributes<HTMLDivElement> & {
+    loading: boolean
+    finished: boolean
+    upgradeProxy?: any
+    context: ConnectedWeb3Context
+  }
+
+export const UpgradeProxy: React.FC<UpgradeProxyProps> = (props: UpgradeProxyProps) => {
+  const { context, finished, loading, upgradeProxy, ...restProps } = props
+  const nativeAsset = getNativeAsset(context.networkId)
+  const state = loading ? ButtonStates.working : finished ? ButtonStates.finished : ButtonStates.idle
+
+  return (
+    <Wrapper {...restProps}>
+      <Title>Enable {nativeAsset.symbol}</Title>
+      <DescriptionWrapper>
+        <Description>
+          This permission allows the smart contracts to interact with your {nativeAsset.symbol}.
```

Comment:
`This permission allows Omen smart contracts to interact with your ETH. This has to bedone for each new token.`